features/admin.template-core.v1/components/template-dynamic-form.tsx

```diff
@@ -0,0 +1,251 @@
+    /**
+     * Function to handle form submission.
+     */
+    onFormSubmit: (values: Record<string, unknown>, callback: () => void) => void;
+    /**
+     * Loading status for the wizard.
+     */
+    isLoading: boolean;
+    /**
+     * Prop to indicate whether the form is read-only.
+     */
+    readOnly: boolean;
+}
+
+/**
+ * Template Dynamic form component.
+ *
+ * @param Props - Props to be injected into the component.
+ */
+export const TemplateDynamicForm: FunctionComponent<TemplateDynamicFormPropsInterface> = ({
+    customInitializers,
+    customSubmissionHandlers,
+    customValidations,
+    form,
+    initialFormValues,
+    templatePayload,
+    buttonText,
+    onFormSubmit,
+    isLoading,
+    readOnly,
+    ["data-componentid"]: componentId = "template-dynamic-form"
+}: TemplateDynamicFormPropsInterface): ReactElement => {
+
+    const { validate } = useValidationHandlers(customValidations);
+    const { initialize } = useInitializeHandlers(customInitializers);
+    const { submission } = useSubmissionHandlers(customSubmissionHandlers);
+
+    const { UIConfig } = useUIConfig();
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const [ isSubmitting, setIsSubmitting ] = useState(false);
+    const [ formInitialValues, setFormInitialValues ] = useState<{ [key: string]: any }>(null);
+
+    /**
+     * Moderate the initially provided data for the form.
+     */
+    useEffect(() => {
+        const prepareInitialValues = async (): Promise<void> => {
+            let initialValues: Record<string, unknown>;
+
+            if (form?.submitDefinedFieldsOnly) {
+                const paths: string[] = form?.fields?.map((field: DynamicFieldInterface) => field?.name);
+
+                initialValues = pick(initialFormValues, paths);
+            } else {
+                initialValues = cloneDeep(initialFormValues);
+            }
+
+            await initialize(initialValues, form?.fields, templatePayload);
+
+            setFormInitialValues(initialValues);
+        };
+
+        if (!initialFormValues || !form || !templatePayload) {
+            return;
+        }
+
+        prepareInitialValues();
+    }, [ initialFormValues, form, templatePayload ]);
+
+    /**
+     * Callback function triggered when clicking the form submit button.
+     *
+     * @param values - Submission values from the form fields.
+     */
+    const onSubmit = async (values: Record<string, unknown>): Promise<void> => {
+        setIsSubmitting(true);
+        const formValues: Record<string, unknown> = cloneDeep(values);
+
+        /**
+         * Make sure that cleared text fields are set to an empty string.
+         */
+        form?.fields?.forEach((field: DynamicFieldInterface) => {
+            if (!has(formValues, field?.name)) {
+                const initialValue: any = get(formInitialValues, field?.name);
+
+                if (initialValue && typeof initialValue === "string") {
```

Comment:
Check the number field as well

Reply:
Currently no support to add a number field in template => Only support Text, TextArea, Checkbox, and FilePicker

As our need when we are adding the number support, we can improve this section.